### ISSUE TITLE
Update Dictionary Version to 1.14 on ARGO Homepage

### ISSUE THREAD

**Issue by b-f-chan** [icgc-argo/platform-ui#2318]
- Update dictionary version to 1.14 on ARGO homepage

So the new announcement will be as follows:
- [x] add the new June 8 sentence above the March 7 item. 
- [x] Link for "Surgery clinical table" can be right to the dictionary page: https://docs.icgc-argo.org/dictionary
- [x] Remove "on the ICGC ARGO Data Platform" from the March 7 item to make is shorter
- [x] Change the version in the last sentence from 1.13 to 1.14

**Announcements:**
`June 8, 2022: Data submitters can now submit supplemental surgery data for donors that have this specified treatment in the new Surgery clinical table. `
March 7, 2022: We are excited to announce Data Release 5.0, which includes Open Access Variant Filtered data in the File Repository for APGI-AU, PACA-CA, OCCAMS-GB, PTC-SA, and LUCA-KR.
**New Features:**
Program dashboards and the File Repository have been updated with new features to filter on Open Access Variant Filtered data, and new file detail pages are now accessible from the File Repository table.
`ARGO Data Dictionary 1.14`, Data Release 5.0 and Software Release 1.105.3 - API 3.28.2 are now available.

![image](https://user-images.githubusercontent.com/18424138/172190377-c82179ea-97ab-4504-9abb-3431c5dceb47.png)

**NOTE:** Takes priority over critical alerts epic (https://github.com/icgc-argo/platform-ui/issues/2208) to enable Dictionary 1.14 release

**Comment by b-f-chan** [comment]
@samrichca was this tested in QA?

**Comment by b-f-chan** [comment]
Tested in QA, all 3 changes successful, however, I noticed a separate issue, not sure if it is intentional or just due to the QA environment, however I notice this discrepancy in prod also.

The software release version and API version in the `Latest News` area is not consistent/same as what is in the footer:

![image.png](https://images.zenhubusercontent.com/5c8bff46913aca2929d7074d/073e9417-d6d0-46a4-b015-59dc58bfe694)

**Comment by b-f-chan** [comment]
Spoke to @samrichca about the version discrepancy, `Latest News` is not dynamic and manually updated, so we should update to reflect the correct versions going out.

Software Release: 1.110.0
API Version: 3.32.0

**Comment by b-f-chan** [comment]
@demariadaniel will update the software + API version in a separate ticket since he's making last change; @samrichca this ticket is tested and looks good, moving to awaiting release

**Comment by samrichca** [comment]
Small addendum: UI is version 1.110.1 due to today's patch

**Comment by kcullion** [comment]
the copy looks good on prod, the links will be correct once dictionary 1.14 is in place
